packages/components/checkbox/index.js

```diff
@@ -71,47 +77,49 @@ export const Checkbox = ({
       ]),
     },
     label: {
-      ml: isReversed ? '0' : '2',
-      mr: isReversed ? '2' : '0',
-      fontFamily: 'body',
-      fontSize: 'checkbox',
-      fontWeight: 'medium',
-      letterSpacing: 'checkbox',
-      color: `${
-        isDisabled ? 'font.checkbox.disabled' : 'font.checkbox.default'
-      }`,
+      ml: isReversed ? '0' : label.normal.margin,
+      mr: isReversed ? label.normal.margin : '0',
+      fontFamily: label.normal.fontFamily,
+      fontSize: label.normal.fontSize,
+      fontWeight: label.normal.fontWeight,
+      letterSpacing: label.normal.letterSpacing,
+      color: `${isDisabled ? label.disabled.color : label.normal.color}`,
       ...getAllowedProps(override.label, [...color, ...space, ...typography]),
     },
   };
 
   return (
-    <Box as="label" {...Override.wrapper}>
-      {/* This is the sibling input, it's visually hidden */}
-      <VisuallyHidden
-        as="input"
-        type="checkbox"
-        id={id}
-        name={name}
-        value={value}
-        checked={isChecked}
-        disabled={isDisabled}
-        onBlur={onBlur}
-        onFocus={onFocus}
-        onChange={onChange}
-        defaultChecked={defaultIsChecked}
-        {...Override.visuallyHidden}
-      />
+    <ThemeProvider theme={theme}>
+      <Box p="2">
+        <Box as="label" {...Override.wrapper}>
+          {/* This is the sibling input, it's visually hidden */}
+          <VisuallyHidden
+            as="input"
+            type="checkbox"
+            id={id}
+            name={name}
+            value={value}
+            checked={isChecked}
+            disabled={isDisabled}
+            onBlur={onBlur}
+            onFocus={onFocus}
+            onChange={onChange}
+            defaultChecked={defaultIsChecked}
```

Comment:
I am confused by `defaultIsChecked`. If we are controlling it through `value` do we really need defaultIsChecked as a prop ? Still we need to say to input to that if it is defaultChecked to make it active state. But whether through a prop is my question.

Reply:
```suggestion
       value={value}
       checked={isChecked}
       disabled={isDisabled}
       defaultChecked={defaultIsChecked}
```
All the above are native HTML attributes, each serve different purpose.
We can control the `checked` with `isChecked` and `onChange`.
But there will be time, where we want the checkbox to be checked by default when we don't want to control `checked` state.
This will be used in simple checkboxes where we won't control the checkbox.
This is a port from Chakra Checkbox.
https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/Checkbox/index.js
https://www.w3schools.com/jsref/prop_checkbox_defaultchecked.asp

Reply:
I agree with your point of having defaultIsChecked on the `input` DOM element.  But my point was that `defaultInputChecked={value}` itself directly since it is being controlled ? Even if we are passing the prop `defaultInputChecked` it will be same as the value I believe.

Reply:
When using we will be using `defaultIsChecked` only. 
I guess I did not understand your point.
Can you make a code suggestion or we can discuss on the next call.
I will make a note in our channel.

Reply:
Added to Trello for more discussion on the topic on the the next call.